catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KMLTransformerImpl.java

```diff
@@ -208,6 +221,91 @@ protected Placemark performDefaultTransformation(Metacard entry)
     return kmlPlacemark;
   }
 
+  private void setExtendedData(Placemark placemark, Metacard metacard) {
+    final ExtendedData extendedData = new ExtendedData();
+
+    final Set<AttributeDescriptor> attributeDescriptors =
+        metacard.getMetacardType().getAttributeDescriptors();
+
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      final String attributeName = attributeDescriptor.getName();
+      final Attribute attribute = metacard.getAttribute(attributeName);
+      if (attribute != null) {
+        String attributeValue =
+            convertAttribute(attribute, attributeDescriptor)
+                .toString()
+                .replace("[", "")
+                .replace("]", "");
+        final Data data = getData(attributeName, attributeValue);
+        extendedData.addToData(data);
+      }
+    }
+    placemark.setExtendedData(extendedData);
+  }
+
+  private Data getData(String attributeAlias, String attributeValue) {
+    final Data data = new Data(attributeValue);
+    data.setName(attributeAlias);
+    return data;
+  }
+
+  private static Object convertAttribute(Attribute attribute, AttributeDescriptor descriptor) {
```

Comment:
nit - I think it might be better to return a list of serializables and return a list of size 1 in the case of a single value, just to make method declarations more clear.